frontend/src/components/HomePageButtons.vue

```diff
@@ -0,0 +1,55 @@
+<template>
+  <span>
+    <v-btn :key="`${button.text}-${index}`" v-for="(button, index) in routeButtons" v-bind="button.props">
+      {{ button.text }}
+    </v-btn>
+  </span>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from "vue-property-decorator";
+import { mapActions } from "vuex";
+import { RouteConfig } from "vue-router";
+import { getHomePageRoutes, getRouteChildren } from "@/router";
+
+@Component({
+  name: "HomePageButtons",
+  methods: {
+    ...mapActions("user", ["getUserClaims"])
+  }
+})
+export default class RequireClaimsButtons extends Vue {
+  userClaims: any = {};
+  routeButtons: any[] = [];
+
+  async mounted() {
+    this.userClaims = (await this.getUserClaims()) || {};
+    this.makeButtons(getHomePageRoutes(getRouteChildren(), this.userClaims));
```

Comment:
I'm sure `getHomePageRoutes` can itself get current user claims and give all accessible routes for home page. Simplify the function interface.